### PR TITLE
Update crd guidelines

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Updated CRD guidelines to reflect current implementation:
   - one installation /upgrade Job
   - one ConfigMap
   - unified naming of K8s objects involved in CRD installation / upgrade process

### CODE REVIEW

File: guidelines/technical-guidelines/03-custom-resource-definitions.md [hunk 1]
```diff
@@ -12,96 +12,30 @@ If you use a third-party CRD, apply the [location and file name](#custom-resourc
...
-    "helm.sh/hook-delete-policy": "before-hook-creation, hook-succeeded"
-data:
-  idppresets.authentication.kyma-project.io: |-
-{{.Files.Get "files/crd-idppreset.yaml" | printf "%s" | indent 4}}
-```
+During the initial phase of installation or upgrade, all CRDs from the `files` subdirectory are bundled by and mounted into one [ConfigMap](https://github.com/kyma-project/kyma/blob/master/resources/cluster-essentials/templates/crd-install-config-map.yaml). This ConfigMap is located under [`resources/cluster-essentials/templates`](https://github.com/kyma-project/kyma/tree/master/resources/cluster-essentials/templates), in the same location as the installation and upgrade Job, the ServiceAccount that the Job uses to apply CRDs, and the ClusterRoleBinding which binds the ServiceAccount with the proper ClusterRole for adequate permissions.
 
 ## CRD installation and upgrade
 
+To make the installation process more efficient and maintainable, we decoupled CRDs from charts and store them in the `cluster-essentials` component.
```
Comment (bszwarc) [inline guidelines/technical-guidelines/03-custom-resource-definitions.md:29]:
```suggestion
To make the installation process more efficient and maintainable, we decoupled CRDs from charts and now store them in the `cluster-essentials` component.

```

File: guidelines/technical-guidelines/03-custom-resource-definitions.md [hunk 2]
```diff
@@ -12,96 +12,30 @@ If you use a third-party CRD, apply the [location and file name](#custom-resourc
...
 
-In order to make the installation process more efficient and maintainable, we've decided to decouple CRDs from charts and store them in the `cluster-essentials` component.
-
-All the CRDs are installed or updated in the first step, with a Kubernetes Job that is triggered by the Helm's [pre-install and pre-upgrade](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) hooks.
+All Kyma CRDs are installed or updated at the very beginning by the [Kubernetes Job](https://github.com/kyma-project/kyma/blob/master/resources/cluster-essentials/templates/crd-install-job.yaml) that is triggered by the Helm's [pre-install and pre-upgrade](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) hooks.
```
Comment (bszwarc) [inline guidelines/technical-guidelines/03-custom-resource-definitions.md:31]:
```suggestion
All Kyma CRDs are installed or updated at the very beginning by the [Kubernetes Job](https://github.com/kyma-project/kyma/blob/master/resources/cluster-essentials/templates/crd-install-job.yaml) that is triggered by Helm's [pre-install and pre-upgrade](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) hooks.
```

File: guidelines/technical-guidelines/03-custom-resource-definitions.md [hunk 3]
```diff
@@ -12,96 +12,30 @@ If you use a third-party CRD, apply the [location and file name](#custom-resourc
 
 Place the Kyma CRDs in the `cluster-essentials` Helm chart folder under the `files` subdirectory.
 
+When creating file names, use the **names:singular** format, for example `crontab.crd.yaml`.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/technical-guidelines/03-custom-resource-definitions.md:15]: Yup, I'll clarify that.

**Comment by m00g3n** [comment]
LGTM

**Pull request merged**